src/blob/core.rs

```diff
@@ -573,8 +586,19 @@ impl RawRecords {
             })?;
         self.current_offset += self.record_header_size;
         self.current_offset += header.meta_size();
+        let data =
+        if read_data {
+            buf.resize(header.data_size() as usize, 0);
+            self.file
+                .read_at(&mut buf, self.current_offset)
```

Comment:
`read_exact_at`

Reply:
file is not std File struct, but File struct from pearl, as I understand read_at() already will read exact buf.len() bytes

Reply:
Ok. Then it is alright